js/create-card.js

```diff
@@ -0,0 +1,79 @@
+import {generateAds} from './data.js';
+
+const cardTemplate = document.querySelector('#card').content.querySelector('.popup');
+const mapCanvas = document.querySelector('#map-canvas');
+
+const photo = cardTemplate.querySelector('.popup__photo');
+
+const fragment = document.createDocumentFragment();
+const fragment2 = document.createDocumentFragment();
+
+const createAds = generateAds();
+
+const getHousingType = (type) => {
+  switch (type) {
+    case 'palace':
+      return 'Дворец';
+    case 'flat':
+      return 'Квартира';
+    case 'house':
+      return 'Дом';
+    case 'bungalow':
+      return 'Бунгало';
+    case 'hotel':
+      return 'Отель';
+  }
+};
+
+const createFeatureElement = (element, container) => {
+  element.forEach((item) => {
+    const featuresItem = container.querySelector(`.popup__feature--${item}`);
+    if (featuresItem) {
+      fragment2.append(featuresItem);
+    }
+  });
+
+  container.innerHTML = '';
+  container.append(fragment2);
+};
+
+
+cardTemplate.querySelector('.popup__photos').innerHTML = '';
+
+const createPhotoElement = (element, container) => {
+  element.forEach((value) => {
+    const photoElement = photo.cloneNode(true);
+    photoElement.src = value;
+    container.append(photoElement);
+  });
+};
+
+const getSimilarAds = () => {
+
+  createAds.forEach((element) => {
```

Comment:
Используй деструктуризацию объекта, код ниже станет нагляднее
```suggestion
  createAds.forEach(({ author, offer }) => {
```